include/unistd.h

```diff
@@ -306,6 +306,7 @@ extern "C"
 
 pid_t   vfork(void);
 pid_t   getpid(void);
+pid_t   getpgrp(void);
```

Comment:
#define getpgrp getpid ?

Reply:
this could make confusing, a separate definition make this clear

Reply:
I think it is better to implement this as a function.

Reply:
> I think it is better to implement this as a function.

I tend to agree.  It is not at all obvious to most people that a PID could be equivalent to a process group (PGRP) ID. That is really a consequence of that fact that NuttX does not support process groups, sessions, group leaders.  Without a process group, the PID (which is the ID of the main thread in NuttX) is a reasonable substitute in many cases, but it is not really the same thing.

For a child of the process, the value returned by getpid() is not the group ID.  The parent process ID would be better in that case.  But without process groups, the motivation to create a high fidelity getpgrp() is not high.

[In NuttX, the use of "group" is overloaded.  NuttX groups are not POSIX groups.]

The IDLE task is usually the group leader for all tasks, but it shouldn't be.  NSH, for example, should be a group leader and all of the tasks started by NSH should have the same group ID.  Killing the NSH task should kill all of the tasks in the group (unless they have daemonized).  Kill with a negated process ID should should kill all tasks in the process group.

So I think it would be good to hide these issues from the user.  The getpgrp() function should optimize by a simple relative branch to getpid() so there is no real burden to